### PR TITLE
Handle retry after

### PR DESCRIPTION
**Pull request by stefaneberl**

### CODE REVIEW

File: src/openkit-shared/Core/Communication/BeaconSendingFlushSessionsState.cs [hunk 1]
```diff
@@ -51,12 +51,17 @@ protected override void DoExecute(IBeaconSendingContext context)
...
             context.FinishedAndConfiguredSessions.ForEach(finishedSession =>
             {
-                if (finishedSession.IsDataSendingAllowed)
+                if (!tooManyRequestsReceived && finishedSession.IsDataSendingAllowed)
                 {
-                    finishedSession.SendBeacon(context.HTTPClientProvider);
+                    var statusResponse = finishedSession.SendBeacon(context.HTTPClientProvider);
+                    if (BeaconSendingResponseUtil.IsTooManyRequestsResponse(statusResponse))
+                    {
+                        tooManyRequestsReceived = false;
```
Comment (y0chanan) [inline src/openkit-shared/Core/Communication/BeaconSendingFlushSessionsState.cs]: tooManyRequestsReceived = true

this checks if there has already been too many requests received, if not but the condition occured it is set to true here

File: tests/openkit-shared-tests/Core/Communication/BeaconSendingCaptureOnStateTest.cs [hunk 2]
```diff
@@ -228,7 +271,7 @@ public void MultiplicityIsSetToZeroIfNoFurtherNewSessionRequestsAreAllowed()
             // also ensure that both got a configuration set
             Assert.That(sessionOne.IsBeaconConfigurationSet, Is.True);
             Assert.That(sessionOne.BeaconConfiguration.Multiplicity, Is.EqualTo(0));
-            
+
```
Comment (y0chanan) [inline tests/openkit-shared-tests/Core/Communication/BeaconSendingCaptureOnStateTest.cs]: remove this unnecessariliy introduced newline

**Review by y0chanan — CHANGES_REQUESTED** [review]

File: tests/openkit-shared-tests/Core/Communication/BeaconSendingCaptureOnStateTest.cs (hunk 2 above)
Comment (stefaneberl) [reply tests/openkit-shared-tests/Core/Communication/BeaconSendingCaptureOnStateTest.cs]: Was not really introduced, but was there already (document reformat).

Thanks to your comment, I saw an old issue in the test. :)

**Review by y0chanan — APPROVED** [review]

File: src/openkit-shared/Protocol/HTTPClientHttpClient.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ protected override HTTPResponse GetRequest(string url, string clientIPAddress)
         {
             using (System.Net.Http.HttpClient httpClient = CreateHTTPClient(clientIPAddress))
             {
-                System.Threading.Tasks.Task<System.Net.Http.HttpResponseMessage> responseTask = httpClient.GetAsync(url);
+                var responseTask = httpClient.GetAsync(url);
```
Comment (thomasgrassauer) [inline src/openkit-shared/Protocol/HTTPClientHttpClient.cs:38]: +1 for using var :-)

**Review by thomasgrassauer — APPROVED** [review]

**Pull request merged**